tests/spyre_util.py

```diff
@@ -528,22 +566,42 @@ def create_text_prompt(model: str, min_tokens: int, max_tokens: int) -> str:
     return prompt
 
 
-def create_random_request(
-        request_id: int, num_tokens: int,
-        sampling_params: SamplingParams) -> EngineCoreRequest:
+def create_random_request(request_id: int,
+                          num_tokens: int,
+                          sampling_params: SamplingParams,
+                          from_model_vocab: bool = False,
+                          model: Optional[str] = None) -> EngineCoreRequest:
 
-    return EngineCoreRequest(request_id=str(request_id),
-                             prompt_token_ids=[request_id] * num_tokens,
-                             mm_inputs=None,
-                             mm_hashes=None,
-                             mm_placeholders=None,
-                             sampling_params=sampling_params,
-                             eos_token_id=None,
-                             arrival_time=0,
-                             lora_request=None,
-                             data_parallel_rank=None,
-                             pooling_params=None,
-                             cache_salt=None)
+    tokenizer = AutoTokenizer.from_pretrained(model)
+    if from_model_vocab:
+        assert model is not None, "Prompt requested to be generated from " \
+        "model's vocabulary: need to provide model."
+
+        valid_token_ids = sorted([
+            v for v in tokenizer.vocab.values()
+            if v not in tokenizer.all_special_ids
+        ])
+        prompt_token_ids = random.choices(valid_token_ids, k=num_tokens)
+    else:
+        # start with existing prompts and tokenize them
+        prompts = get_longer_chicken_soup_prompts(4)
```

Comment:
🤔  There's no guarantee that this will give you the number of tokens needed, and given current conversations with the compiler team we may need to add some unit test cases with longer prompts in the near future.

Maybe instead of making this more robust now, we could at least add an assert that the prompt has the correct number of tokens?